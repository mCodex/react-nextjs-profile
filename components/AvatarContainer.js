import React, { PureComponent, Fragment } from 'react';

import { Image, Header, Flag, Button } from 'semantic-ui-react';

import withData from '../hocs/DataProvider';

const styles = {
  container: {
    display: 'flex',
    flex: 1,
    width: '100%',
    height: '50vh',
    backgroundColor: '#eee',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    marginBottom: 20
  },
  flagContainer: {
    display: 'flex',
    width: '100%',
    flexDirection: 'row',
    backgroundColor: '#eee',
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
    padding: 10
  }
};

class AvatarContainer extends PureComponent {
  constructor(props) {
    super(props);
    this.t = props.t;
  }

  render() {
    const { data } = this.props;
    return (
      <Fragment>
        <div style={styles.flagContainer}>
          {data.i18n.map(country =>
            (
              <Button
                basic
                compact
              >
                <Flag name={country} />
              </Button>
           ))
        }
        </div>
        <div style={styles.container}>
          <Image
            src={data.pic}
            size="small"
            circular
          />
          <Header as="h1" style={{ marginTop: 20 }}>{data.fullname}</Header>
        </div>
      </Fragment>
    );
  }
}

export default withData(AvatarContainer);
