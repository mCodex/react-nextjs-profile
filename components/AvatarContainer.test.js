import React from 'react';
import renderer from 'react-test-renderer';

import AvatarContainer from './AvatarContainer';

describe('AvatarContainer component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AvatarContainer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
