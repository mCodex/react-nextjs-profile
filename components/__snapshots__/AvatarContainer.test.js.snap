// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarContainer component renders correctly 1`] = `
Array [
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#eee",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <button
      className="ui basic compact button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        className="br flag"
      />
    </button>
    <button
      className="ui basic compact button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        className="us flag"
      />
    </button>
  </div>,
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eee",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "height": "50vh",
        "justifyContent": "center",
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <img
      className="ui small circular image"
      src="https://semantic-ui.com/images/avatar2/large/matthew.png"
    />
    <h1
      className="ui header"
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      John Doe
    </h1>
  </div>,
]
`;
